MultiplayerSample Project needs to update its reference to the Multiplayer Gem jinja templates

The Multiplayer Sample contains a reference to jinja template files within the Multiplayer Gem that are not available in an SDK layout build.
 The [multiplayersample_autogen_files.cmake](https://github.com/o3de/o3de-multiplayersample/blob/development/Gem/Code/multiplayersample_autogen_files.cmake) is referencing jinja template files from the Engine Source, which doesn't get copied to the Install SDK layout due to being in a private directory.

Furthermore that cmake file hardcodes the directory of the Multiplayer Gem location to be `${LY_ROOT_FOLDER}/Gems/Multiplayer` which isn't portable and would cause the MultiplayerSample project to fail to configure if the Multiplayer Gem was relocated outside of the Engine Source Repo or the folder was renamed.

Instead what should be done is to access the source directory of the Multiplayer Gem, that way the Multiplayer Gem can be relocated without breaking the Multiplayer Sample project
Running the `get_property(multiplayer_gem_source_dir TARGET Multiplayer PROPERTY SOURCE_DIR)` command can be used to lookup the source directory without using a generator expression

> Instead what should be used is a generator expression that can access the source directory of the Multiplayer Gem, that way the Multiplayer Gem can be relocated without breaking the Multiplayer Sample project

But generator expressions are not expanded when assigned to a variable.
I have the following error in my project with the suggested approach:

```
CMake Error at D:/Programs/Windows/o3de-install/cmake/FileUtil.cmake:34 (message):
  File
  D:/Git/gardum/Code/$<TARGET_PROPERTY:Multiplayer,SOURCE_DIR>/Code/Source/AutoGen/AutoComponent_Common.jinja
  referenced in gardum_autogen_files.cmake not found
Call Stack (most recent call first):
  D:/Programs/Windows/o3de-install/cmake/LYWrappers.cmake:102 (ly_include_cmake_file_list)
  Code/CMakeLists.txt:23 (ly_add_target)
```

> > Instead what should be used is a generator expression that can access the source directory of the Multiplayer Gem, that way the Multiplayer Gem can be relocated without breaking the Multiplayer Sample project
> 
> But generator expressions are not expanded when assigned to a variable.
> I have the following error in my project with the suggested approach:
> 
> ```
> CMake Error at D:/Programs/Windows/o3de-install/cmake/FileUtil.cmake:34 (message):
>   File
>   D:/Git/gardum/Code/$<TARGET_PROPERTY:Multiplayer,SOURCE_DIR>/Code/Source/AutoGen/AutoComponent_Common.jinja
>   referenced in gardum_autogen_files.cmake not found
> Call Stack (most recent call first):
>   D:/Programs/Windows/o3de-install/cmake/LYWrappers.cmake:102 (ly_include_cmake_file_list)
>   Code/CMakeLists.txt:23 (ly_add_target)
> ```

Sounds like we would have to go with an alternative call to `get_property(multiplayer_gem_source_dir TARGET Multiplayer PROPERTY SOURCE_DIR)` which will read the target property from the Multiplayer target without needing to go through a generator expression

--- a/Gem/Code/multiplayersample_autogen_files.cmake
+++ b/Gem/Code/multiplayersample_autogen_files.cmake
@@ -5,12 +5,14 @@
 #
 #
 
+get_property(multiplayer_gem_root GLOBAL PROPERTY "@GEMROOT:Multiplayer@")
+
 set(FILES
-    ${LY_ROOT_FOLDER}/Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponent_Common.jinja
-    ${LY_ROOT_FOLDER}/Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponent_Header.jinja
-    ${LY_ROOT_FOLDER}/Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponent_Source.jinja
-    ${LY_ROOT_FOLDER}/Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponentTypes_Header.jinja
-    ${LY_ROOT_FOLDER}/Gems/Multiplayer/Code/Include/Multiplayer/AutoGen/AutoComponentTypes_Source.jinja
+    ${multiplayer_gem_root}/Code/Include/Multiplayer/AutoGen/AutoComponent_Common.jinja
+    ${multiplayer_gem_root}/Code/Include/Multiplayer/AutoGen/AutoComponent_Header.jinja
+    ${multiplayer_gem_root}/Code/Include/Multiplayer/AutoGen/AutoComponent_Source.jinja
+    ${multiplayer_gem_root}/Code/Include/Multiplayer/AutoGen/AutoComponentTypes_Header.jinja
+    ${multiplayer_gem_root}/Code/Include/Multiplayer/AutoGen/AutoComponentTypes_Source.jinja
     
     # Scripting samples
     Source/AutoGen/ScriptingPlayerMovementComponent.AutoComponent.xml

